src/Microsoft.Health.Fhir.Core/Models/SearchParameterInfo.cs

```diff
@@ -15,12 +15,16 @@ public class SearchParameterInfo
             string name,
             Uri url,
             string searchParamType,
-            ICollection<SearchParameterComponentInfo> components)
+            IReadOnlyList<SearchParameterComponentInfo> components,
+            string expression,
+            IReadOnlyCollection<string> targetResourceTypes)
```

Comment:
null check?

Reply:
A lot of these seem to be optional, how do you feel about giving them defaults (i.e. expression = null) or allow property setters?